packages/webapp/src/pages/Connection/List.tsx

```diff
@@ -128,14 +123,23 @@ export default function ConnectionList() {
 
         connectUI.current = nango.openConnectUI({
             baseURL: globalEnv.connectUrl,
-            sessionToken: res.json.data.token,
             onEvent: (event) => {
                 if (event.type === 'close') {
                     // we refresh on close so user can see the diff
                     void mutate();
                 }
             }
         });
+
+        // We defer the token creation so the iframe can open and display a loading screen
+        //   instead of blocking the main loop and no visual clue for the end user
+        setTimeout(async () => {
+            const res = await apiConnectSessions(env);
+            if ('error' in res.json) {
+                return;
+            }
+            connectUI.current!.setSessionToken(res.json.data.token);
+        }, 10);
```

Comment:
that's a pretty short value. no? what happen if the call to `connect/sessions` takes longer than that?

Reply:
The call is triggered after 10ms.

The setTimeout is just here to free the main event loop, which means the iframe can be created and the URL loading started, the iframe doesn't have a separate sandbox so it's important that it has time to load and display something. 
Then the setTimeout triggers and it takes as much time as it needs. 
10ms is not a lot but on a good connection waiting for nothing would be detrimental to UX.